pkg/applicationserver/linking.go

```diff
@@ -226,10 +252,8 @@ func (as *ApplicationServer) link(ctx context.Context, ids ttnpb.ApplicationIden
 	registerLinkStart(ctx, l)
 	go func() {
 		<-ctx.Done()
-		if err := ctx.Err(); errors.IsCanceled(err) {
-			logger.Info("Unlinked")
-			registerLinkStop(ctx, l)
-		}
+		logger.WithError(ctx.Err()).Info("Unlinked")
```

Comment:
We need to always count the link stops, no matter if they ended in an error or not. Otherwise we cannot determine how many links are active at a time (since errors are logged even before the link starts - for connection failure purposes).